### DISCUSSION TITLE
Multiple SLI providers per SLO

### DISCUSSION THREAD

**Discussion by ghost** [keptn/keptn#4672] [Q&A]
This may already be possible and I'm sure I've seen it discussed - but a search both here an on [Slack](https://slack.keptn.sh) yielded nothing so raising here for discussion.

The ability to pull SLIs from multiple providers and evaluate in a single quality gate would be advantageous. I imagine the SLI to provider mapping would live in the sli.yaml file which leaves the slo.yaml as platform independent.

Conceptually though, here's what I imagine it'd look like:

**slo.yaml**
```
spec_version: '1.0'
comparison:
  compare_with: "several_results"
  include_result_with_score: "pass"
  number_of_comparison_results: 3
  aggregate_function: avg
objectives:
- sli: disk_reads
  displayName: "Disk Reads"
  pass:
  - criteria:
    - "<=+2%"
- sli: response_time_p50
  displayName: "Response Time P95"
  pass:
  - criteria:
    - "<=+10%"
    - "<600"
total_score:
  pass: "90%"
  warning: "75%"
```

Where `disk_reads` comes from `sli provider A` and `response_time_p50` comes from `sli provider B`.

**Comment by johannes-b** [comment]
Hi Adam, 
due to an architectural debt, it is not possible to plug in multiple SLI-provider. The changes that are needed to support this use-case I would like to summarize this week. I`ll keep you posted on that, by referencing the enhancement proposal.

**Reply by johannes-b** [reply]
Hi Adam, 
today I summarized the current situation that needs to be addressed to support your use-case. Please take a look here: https://github.com/keptn/enhancement-proposals/pull/56
